adding support for passing cookies to confluence

lalkaka opened the pull request:
using the CONFLUENCE_COOKIES env var

Commit:
adding support for passing cookies to confluence through CONFLUENCE_COOKIES env var

Commit:
adding debug info and allowing the publishing of blog posts

parente commented:
@lalkaka are you still interested in getting this PR into the code base? Would you mind sharing your use case for passing cookies (e.g., what cookies and for what purpose?) so I can better understand if this is a generally useful feature?

lalkaka commented:
Yes, still interested! My company's Confluence instance is on-prem and sits
behind https://www.cloudflare.com/products/cloudflare-access/, which allows
cookies to be passed to get past the SSO interstitial. I believe other
similar zero-trust app gateways (eg from zscaler, Akamai, etc), function in
similar ways.

On Tue, Mar 12, 2019 at 6:30 AM Peter Parente <notifications@github.com>
wrote:

> @lalkaka <https://github.com/lalkaka> are you still interested in getting
> this PR into the code base? Would you mind sharing your use case for
> passing cookies (e.g., what cookies and for what purpose?) so I can better
> understand if this is a generally useful feature?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Valassis-Digital-Media/nbconflux/pull/23#issuecomment-472000794>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAm_U5ZV0sM00SIc5DGmZyf8oYgE8nb7ks5vV6vzgaJpZM4af5A7>
> .
>

Commit:
Add --no-prompt flag to avoid getting username and pass via prompt

This allows cookies to work as a lone authorization method whereas
previously it was required that you entered a username and password
every time. Now it will simply attempt to load all three of username,
password and cookies from environment variables and use what it gets.
The gist of it is that any credentials of the 3 total provided through
environment variables or saved in config must be correct, but you are
only required to give either username and pass OR cookies.

lalkaka committed:
Merge pull request #1 from Kbman99/NoPromptFlag

Add --no-prompt flag to avoid getting username and pass via prompt
